README.md

```diff
@@ -919,6 +919,42 @@ parsers, it may be more convenient to use the `xml_mode` context manager:
     # ... We have now returned to using the default XML parser.
 ```
 
+Application Cleanup
+-------------------
+
+A user-defined function may be registered to be invoked before, during, or after the time
+when the Python iRODS Client is engaged in object teardown (also called "cleanup") operations.
+This is analogous to the atexit facility in C or in Python itself, except that we have the extra
+resolution to specify that a function or callable object be run either just before, or just after, the
+aforementioned object teardown stage:
+
+```
+    from irods import at_client_exit
+    at_client_exit.register_for_execution_after_prc_cleanup(lambda : print("PRC cleanup has completed."))
+    at_client_exit.register_for_execution_before_prc_cleanup(lambda : print("PRC cleanup is about to start."))
+```
+Alternatively, we can use a lower-level interface to access the "during" aspect of things and run a function closer in
+time to the actual Python client cleanup processing.
```

Comment:
Not sure if I asked this before, but what does this achieve over the `before` phase/stage?

Does it cover some very niche use-case?

Reply:
No, I can't really imagine many uses for DURING outside of PRC itself in an application.  Possibly for debug I guess, where you might invoke a print or a `pdb` breakpoint via `set_trace` to inspect things right before PRC teardown but after the BEFORE stage so that presumably higher abstractions are possibly gone.

Reply:
Got it.